pkg/comp-functions/functions/vshnkeycloak/deploy.go

```diff
@@ -235,12 +236,47 @@ func handleCustomConfig(ctx context.Context, comp *vshnv1.VSHNKeycloak, svc *run
 	lastConfigHash := comp.GetLastConfigHash()
 	lastEnvHash := comp.GetLastEnvHash()
 
+	// If we just created the instance, then the last hashes will be empty and thus trigger configOrEnvChanged().
+	// This will result in a race condition where two keycloak setups could run in parallel and clash with each other.
+	// Therefore, we'll check if those hashes are empty and set them to the current hashes.
+	if (lastConfigHash == "" && currentConfigHash != "") || (lastEnvHash == "" && currentEnvHash != "") {
```

Comment:
Did you test this change for the config apply job in Lab? If not I can help you out, I have all the dummy configs at my fingertips